fusillade/resource.py

```diff
@@ -0,0 +1,225 @@
+                             object=dict(
+                                 type=new_node.object_type,
+                                 path_name=new_node._path_name
+                             ),
+                             policy=dict(
+                                 link_name='Owner',
+                                 policy_type='IAMPolicy')
+                             ))
+            return new_node
+
+    # TODO: Add function to modify actions, this will need to modify all access policies with this actions
+    @property
+    def actions(self):
+        if not self._actions:
+            self._get_attributes(self._attributes)
+        if isinstance(self._actions, str):
+            self._actions = self._actions.split(' ')
+        return self._actions
+
+    def add_actions(self):
+        set(self._actions)
+
+    def check_actions(self, policy: dict):
+        policy_actions = set()
+        for s in policy['Statement']:
+            policy_actions.update(s['Action'])
+        if not policy_actions.issubset(set(self.actions)):
+            raise FusilladeBadRequestException(detail="Invalid actions in policy.")
+
+    @staticmethod
+    def hash_name(name):
+        return name
+
+    def list_policies(self, per_page=None, next=None):
+        children, next_token = self.cd.list_object_children_paged(f"{self.object_ref}/policy", next, per_page)
+        return [f"/resource/{self.name}/policy/{child}" for child in children.keys()], next_token
+
+    @staticmethod
+    def format_policy(policy: dict) -> str:
+        policy.update(Version="2012-10-17")
+        return json.dumps(policy)
+
+    def create_policy(self,
+                      policy_name: str,
+                      policy: dict,
+                      policy_type: str = 'IAMPolicy',
+                      parent_path: str = None,
+                      run=True,
+                      **kwargs) -> Union[None, List[Dict[str, Any]]]:
+        """
+        Create a policy object and attach it to the ResourceType.
+
+        :param policy_name:
+        :param policy: Json string that follow AWS IAM Policy Grammar.
+          https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_grammar.html
+        :param policy_type: Either 'IAMPolicy' or 'ResourcePolicy'.
+        :param parent_path: The cloud directory path to the parent node the policy will be attached. If None the
+        policy is attached to /resource/{self.name}/policy
+        :param run:
+        :param kwargs: additional attributes describing the policy.
+        :return:
+        """
+        if policy_type == "ResourcePolicy":
+            self.check_actions(policy)
+        operations = list()
+        object_attribute_list = self.cd.get_policy_attribute_list(policy_type, self.format_policy(policy), **kwargs)
+        parent_path = parent_path or f"{self.object_ref}/policy"
+        batch_reference = f'new_policy_{policy_name}'
+        operations.append(
```

Comment:
Is `operations` always a list with a single item?

Reply:
If run is False, `operations` are returned and can be combined with other operations to perform a batch request. In this specific function only one operation is added.